MessagingServiceTests/MessagingServicePOSTTests.cs

```diff
@@ -0,0 +1,975 @@
+            {
+                var startup = new Startup();
+                startup.ConfigureAADManager(aadManager);
+                startup.Configuration(app);
+            });
+            */
+
+        }
+
+        [ClassCleanup]
+        public static void ClassClean()
+        {
+            _server.Dispose();
+            _server_noStorage.Dispose();
+        }
+
+
+        [TestCleanup]
+        public async void Cleanup()
+        {
+            bool result = await _storageManager.Delete("email1");
+        }
+
+
+        /// <summary>
+        /// Tests happy path.
+        /// </summary>
+        [TestMethod]
+        public async Task TestContactOwners()
+        {
+            string fileContent = File.ReadAllText(TestJSONPath);
+            StringContent postContent = new StringContent(fileContent, Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await _server.HttpClient.PostAsync("/contactOwners", postContent);
+            Stream errors = response.Content.ReadAsStreamAsync().Result;
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                StreamReader errorsReader = new StreamReader(errors);
+                string errorsString = errorsReader.ReadToEnd();
+                JObject errorsJSON = JObject.Parse(errorsString);
+            }
+
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+
+
+            // Check message
+            StreamStorageContent messageJSON = (StreamStorageContent)await _storageManager.Load("email1");
+            StreamReader reader = new StreamReader(messageJSON.GetContentStream());
+            string bodyContent = reader.ReadToEnd();
+            JObject root = JObject.Parse(bodyContent);
+
+            Assert.AreEqual("user1@gmail.com,user2@gmail.com,user3@gmail.com", root["to"]);
+            Assert.AreEqual("someuser@live.com", root["from"]);
+            Assert.AreEqual("someuser@live.com", root["cc"]);
+            Assert.AreEqual("[PowerShellGallery] Message for owners of the module 'SomeTestPackage'", root["subject"]);
+            Assert.AreEqual(@"User rebro-1 &lt;someuser@live.com&gt; sends the following message to the owners of module 'SomeTestPackage':
+            
+            Hello owners, I would like to be an owner too.  Please add me!
+
+    -----------------------------------------------
+    To stop receiving contact emails as an owner of this module, sign in to the PowerShell Gallery and 
+    [change your email notification settings](/profile/edit).", root["body"]["text"]);
+            Assert.AreEqual(@"User rebro-1 &lt;someuser@live.com&gt; sends the following message to the owners of module 'SomeTestPackage':
+            
+            Hello owners, I would like to be an owner too.  Please add me!
+
+    -----------------------------------------------
+    <em>
+    To stop receiving contact emails as an owner of this module, sign in to the PowerShell Gallery and 
```

Comment:
*Disclaimer: I haven't reviewed the code. Can you verify these are accounted for though?*

Note that we used Markdown before and *always* processed the whole body into HTML.  This led to a couple of problems:

1. When there were URLs with underscores or other Markdown-y characters, it would get encoded when it shouldn't
2. Users that could only read plain text email only get HTML and it was ugly

I'm good with Markdown if we can limit the processing to only the static text we know is Markdown and skip the Markdown processing over data fields and whatnot.

We should also have the alternate stream for plain text.